analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/DeconstructionAssignment.cs

```diff
@@ -0,0 +1,89 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class DeconstructionAssignment : SimpleProcessor<IDeconstructionAssignmentOperationWrapper>
+{
+    protected override IDeconstructionAssignmentOperationWrapper Convert(IOperation operation) =>
+        IDeconstructionAssignmentOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState Process(SymbolicContext context, IDeconstructionAssignmentOperationWrapper assignment)
+    {
+        var operationValues = CollectOperationValues(context.State, assignment.Target, assignment.Value);
+        var newState = context.State;
+        foreach (var (tupleMember, value) in operationValues)
+        {
+            newState = newState.SetOperationAndSymbolValue(tupleMember, value);
+        }
+        return newState;
+    }
+
+    private static IEnumerable<OperationValue> CollectOperationValues(ProgramState state, IOperation target, IOperation value)
+    {
+        var operationValues = new List<OperationValue>();
+        var leftTupleElements = TupleElements(target, state);
+        // If the right side is a tuple, then every symbol/constraint is copied to the left side.
+        if (Unwrap(value, state).AsTuple() is { } rightSideTuple)
+        {
+            var rightTupleElements = TupleElements(rightSideTuple.WrappedOperation, state);
+            for (var i = 0; i < leftTupleElements.Length; i++)
+            {
+                var leftTupleMember = leftTupleElements[i];
+                var rightTupleMember = rightTupleElements[i];
+                if (leftTupleMember.Kind == OperationKindEx.Discard)
+                {
+                    continue;
+                }
+                if (leftTupleMember.AsTuple() is { } nestedTuple)
+                {
+                    operationValues.AddRange(CollectOperationValues(state, nestedTuple.WrappedOperation, rightTupleMember));
+                }
+                else
+                {
+                    operationValues.Add(new(leftTupleMember, state[rightTupleMember]));
+                }
+            }
+        }
+        // If the right side is not a tuple, then every member of the left side tuple is set to empty.
+        else
+        {
+            operationValues.AddRange(leftTupleElements
+                .Where(x => x.Kind != OperationKindEx.Discard)
+                .Select(x => new OperationValue(x, SymbolicValue.Empty)));
+        }
+        return operationValues;
+    }
+
+    private static IOperation[] TupleElements(IOperation operation, ProgramState state) =>
```

Comment:
I looked at this again: The second use of this method is `TupleElements(rightSideTuple.WrappedOperation, state)`. In this case the `Unwrap` is unnecessary. For the other case it can easily be done outside of the method. I think doing so and changing this method to take a `ITupleOperationWrapper` as a parameter would make this nice. Especially if you then also change the type of the `target` parameter in `CollectOperationValues`.